rest-service/manager_rest/rest/resources_v3_1/deployments.py

```diff
@@ -540,18 +567,23 @@ def _set_group_attributes(self, sm, group, request_dict):
             group.default_blueprint = sm.get(
                 models.Blueprint, request_dict['blueprint_id'])
 
-    def _set_group_deployments(self, sm, group, request_dict):
+    def _set_group_deployments(self, sm, group, request_dict, clear=True):
         deployment_ids = request_dict.get('deployment_ids')
         if deployment_ids is not None:
             deployments = [sm.get(models.Deployment, dep_id)
                            for dep_id in deployment_ids]
-            group.deployments.clear()
+            if clear:
```

Comment:
This means that if clear is true but no new deployment IDs were passed in, the group won't be cleared. Should that be the case?

Reply:
indeed. The point is the converse: if deployment ids were passed in, we need both a function to replace (PUT) and append (PATCH).
If deployment ids were not passed in, nothing to do, because at most we're only appending.

Note that indeed this means PUTing an object which does not contain deployment_ids, will not clear the group. That is deliberate, there's a test that checks this (`test_update_description`), and this is because well, it's just seems more useful to not clear the group (a very destructive operation) unless _very explicitly_ asked for it

Reply:
now that I look at it, that test doesn't actually have the assert that it suggests it would have (in the docstring). So I'm adding that assert.

Reply:
OK, sounds reasonable.